### PR TITLE
Data warehouse enrichment endpoint

### PR DESCRIPTION
**Pull request by Kevin-Chant**
### Features and Changes

Creates a new API endpoint for serving information about all configured SdkConnections. This is only accessible by superAdmins as it provides cross-org information, and will be used to provide a mapping from SdkConnection ID to organization & datasource IDs for the data ingestion service to consume.

### Testing

1. Create a Personal Access Token for your superadmin user
2. Run the following script, replacing the placeholder with your PAT
```bash
#!/bin/bash
curl \
  --header "Authorization: Bearer <Personal_Access_Token>" \
  --request GET \
  http://localhost:3100/api/v1/ingestion/data-enrichment
```

### CODE REVIEW

File: packages/back-end/src/api/openapi/openapi.yaml
Comment (romain-growthbook) [inline packages/back-end/src/api/openapi/openapi.yaml:113]: I don't think that you need this to be part of the documented public API.

File: packages/back-end/src/api/openapi/openapi.yaml
Comment (romain-growthbook) [inline packages/back-end/src/api/openapi/openapi.yaml:294]: Same

File: packages/back-end/src/api/openapi/paths/getDataEnrichment.yaml
Comment (romain-growthbook) [inline packages/back-end/src/api/openapi/paths/getDataEnrichment.yaml:1]: Same

File: packages/back-end/src/api/openapi/openapi.yaml
Comment (Kevin-Chant) [reply packages/back-end/src/api/openapi/openapi.yaml:113]: Yeah I was thinking about removing this & the tag on the new endpoint but wasn't sure what our exact stance is

**Review by romain-growthbook — COMMENTED** [review]
I don't think that this should be part of the public API for now. You can define functions outside of the `/api` path. See for instance `src/controllers/slack.ts` for a simple example.

File: packages/back-end/src/api/ingestion/ingestion.router.ts
Comment (romain-growthbook) [inline packages/back-end/src/api/ingestion/ingestion.router.ts:43]: nit: I know JS allows functions to be anywhere in the modules but I think keeping functions declarations before their call promotes much more readable code..

File: packages/back-end/src/controllers/ingestion.ts
Comment (romain-growthbook) [inline packages/back-end/src/controllers/ingestion.ts:32]: Shall this be cached?

**Review by romain-growthbook — APPROVED** [review]
This looks good to me. We should make sure to validate the approach with another eng before merging.

File: packages/back-end/src/controllers/ingestion.ts
Comment (Kevin-Chant) [inline packages/back-end/src/controllers/ingestion.ts:24]: There's also a cache on the ingestion endpoint which will ping this every time there's a cache miss but otherwise won't update its data

**Pull request merged**
